soroban-env-host/src/auth.rs

```diff
@@ -1,3 +1,151 @@
+//! # Auth conceptual overview
+//!
+//! This module is responsible for two separate tasks (both starting with
+//! "auth"):
+//!
+//!   - Authorization: deciding if some action should be allowed by some policy.
+//!   - Authentication: deciding if some credential or signature is authentic.
+//!
+//! As one would expect, authorization can (though doesn't always) depend on
+//! authentication: part of judging whether some action is allowed may depend on
+//! someone presenting a signed credential, at which point one must evaluate the
+//! credential's authenticity.
+//!
+//! Moreover this subsystem is responsible (as will be discussed in detail
+//! below) with facilitating two different _directions_ for each of these tasks:
+//!
+//!   - Contracts can _require_ auth services provided by this module.
+//!   - Contracts can _provide_ auth services required by this module.
+//!
+//! And again, in both directions the "auth services" required or provided may
+//! be _either_ of authorization, authentication, or both.
+//!
+//! All auth services reason about invocations and authorizations, so we
+//! next turn our attention to these.
+//!
+//! ## Invocations
+//!
+//! A Soroban transaction can be seen as a tree of _invocations_: usually
+//! invocations of contract functions, though strictly speaking transctions
```

Comment:
We actually may have host functions in the middle of the invocation tree now (specifically, 'create contract' host fn). So we may have call stacks like `contract->create contract host fn->__check_auth->other contract` etc. So the generalized behavior is that call tree has a mix of host function and contract calls, but most of the host function calls are transparent for auth and are omitted for simplicity.

Reply:
TIL! I'll update that to clarify, thanks.